src/libLookup/Lookup.cpp

```diff
@@ -1901,6 +2085,69 @@ bool Lookup::ProcessSetStateDeltaFromSeed(const bytes& message,
   return true;
 }
 
+bool Lookup::ProcessSetStateDeltasFromSeed(const bytes& message,
+                                           unsigned int offset,
+                                           const Peer& from) {
+  LOG_MARKER();
+
+  if (AlreadyJoinedNetwork()) {
+    cv_setStateDeltasFromSeed.notify_all();
+    return true;
+  }
+
+  unique_lock<mutex> lock(m_mutexSetStateDeltasFromSeed);
+
+  uint64_t lowBlockNum = 0;
+  uint64_t highBlockNum = 0;
+  vector<bytes> stateDeltas;
+  PubKey lookupPubKey;
+
+  if (!Messenger::GetLookupSetStateDeltasFromSeed(message, offset, lowBlockNum,
+                                                  highBlockNum, lookupPubKey,
+                                                  stateDeltas)) {
+    LOG_EPOCH(WARNING, m_mediator.m_currentEpochNum,
+              "Messenger::GetLookupSetStateDeltasFromSeed failed.");
+    return false;
+  }
+
+  if (!VerifySenderNode(GetSeedNodes(), lookupPubKey)) {
+    LOG_EPOCH(WARNING, m_mediator.m_currentEpochNum,
+              "The message sender pubkey: "
+                  << lookupPubKey << " is not in my lookup node list.");
+    return false;
+  }
+
+  LOG_EPOCH(INFO, m_mediator.m_currentEpochNum,
+            "ProcessSetStateDeltasFromSeed sent by "
+                << from << " for blocks: " << lowBlockNum << " to "
+                << highBlockNum);
+
+  for (const auto& delta : stateDeltas) {
+    if (!AccountStore::GetInstance().DeserializeDelta(delta, 0)) {
+      LOG_GENERAL(WARNING,
+                  "AccountStore::GetInstance().DeserializeDelta failed");
+      return false;
+    }
+  }
+
+  bool isAnyTxBlkFromVacuousEpoch = false;
+  for (auto i = lowBlockNum; i <= highBlockNum; i++) {
+    if ((i + 1) % NUM_FINAL_BLOCK_PER_POW == 0) {
+      isAnyTxBlkFromVacuousEpoch = true;
+      break;
+    }
+  }
+
+  if (isAnyTxBlkFromVacuousEpoch &&
```

Comment:
This judge is a bit weird. E.g. a node trying to sync at epoch 21 (10 tx epoch / ds), the whole states it fetched is at epoch 10 (very likely to happen as the states uploading may not finish). Then the state delta is from epoch 11 ~ 21. Which includes an vacuous epoch. Then in this case, we will result in calling `MoveUpdatesToDisk` for the state in epoch 21, this is what we don't want.
So the change I suggest is to call `MoveUpdatesToDisk` only when the highBlockNum being the vacuous block.

Reply:
Agree.